test/celeritas/grid/CalculatorTestBase.cc

```diff
@@ -50,8 +51,13 @@ auto CalculatorTestBase::mutable_values() -> SpanReal
 /*!
  * Construct from an arbitrary function.
  */
-void CalculatorTestBase::build(Real2 bounds, size_type count, XsFunc calc_xs)
+void CalculatorTestBase::build(Real2 bounds,
+                               size_type count,
+                               XsFunc calc_xs,
+                               bool spline)
```

Comment:
Instead of `bool spline` (very difficult to tell what's `true` in the caller) how about making this an implementation function

```suggestion
void CalculatorTestBase::build_impl(Real2 bounds,
                               size_type count,
                               XsFunc calc_xs,
                               SplineDerivCalculator::BoundaryCondition bc)
```
and build the spline data if `bc != SplineDerivCalculator::BoundaryCondition::size_`, then adding two inline functions in the header:

```c++
void build(Real2 bounds,
                               size_type count,
                               XsFunc calc_xs)
{
return this->build(bounds, count, std::move(calc_xs), BC::size_);
}
void build_spline(Real2 bounds,
                               size_type count,
                               XsFunc calc_xs,
BC bc)
{
CELER_EXPECT(bc != BC::size_);
return this->build(bounds, count, std::move(calc_xs), bc);
}
```